packages/eslint-plugin/src/index.ts

```diff
@@ -29,8 +34,25 @@ export const rules = {
   'no-suppress-xcss': noSuppressXCSS,
   'no-empty-styled-expression': noEmptyStyledExpressionRule,
   'shorthand-property-sorting': shorthandFirst,
-};
+} as const;
+
+export const plugin = {
+  name,
+  version,
+  rules,
+  configs: {
+    recommended,
+    'flat/recommended': {
+      ...flatRecommended,
+      plugins: {
+        get '@compiled'() {
+          return plugin;
+        },
```

Comment:
This is a bit more coherent than `plugin.configs['flat/recommended'].plugins['@compiled'] = plugin` which is what you'd have to do otherwise, right?

Only real concern is could this ever backfire with some Node support?  I feel like it's been around _forever_ but also a syntax I never use (because when it was introduced it wasn't ready to be used)…it's good now, right?

Also, would you ever want to merge this though?
```suggestion
        ...flatRecommended.plugins,
        get '@compiled'() {
          return plugin;
        },
```